src/librustc_typeck/astconv.rs

```diff
@@ -1277,8 +1277,18 @@ impl<'o, 'gcx: 'tcx, 'tcx> dyn AstConv<'gcx, 'tcx>+'o {
         // Find the type of the associated item, and the trait where the associated
         // item is declared.
         let bound = match (&ty.sty, ty_path_def) {
+            (&ty::Adt(adt_def, _), Def::TyAlias(_)) if adt_def.is_enum() => {
```

Comment:
`Def::TyAlias(_)` shouldn't be here, we only look at the semantic type and check whether it's enum.